### DISCUSSION TITLE
AWS ElastiCache / Redis setup

### DISCUSSION THREAD

**Discussion by peteror** [saleor/saleor#10698] [Deployment] [answered]
I'm running a Saleor store for the past few months on AWS. (Postgres on RDS, docker images deployed on ECS) 
A bit after the start I added Redis (on ElastiCache), but it did not seem to affect performance. On ElastiCache metrics I see items in the cache and connections to Redis, but the cache hit rate is 0.

I set the cache in the CACHE_URL environment variable, like : CACHE_URL=redis://redis-prod.********.cache.amazonaws.com:6379/0 

ElastiCache and the Saleor containers are in the same VPC and security group. 
I tried to use Elasticache for celery broker too - I set the environment variable, but all the callbacks stopped (nothing in the logs though).

My front-end is a mobile app that uses the GraphQL APIs directly. 

Is there anything else I should check / do to make caching work?

**Comment by fowczarek** [comment, marked as answer]
Saleor doesn't use cache in many places. Mainly cache is used to optimize plugins like `Avatax` or `OpenId`.  Another place that uses cache is sync webhook for excluding shipping method. Cach is also used to optimize introspection queries.
To confirm that cache works you need to perform the same introspection query at least 2 times. **Introspection query is cached only if debug mode is turned off.** 

Sample IntrospectionQuery:
```
query IntrospectionQuery {
  __schema {
    queryType {
      name
    }
    mutationType {
      name
    }
    subscriptionType {
      name
    }
    types {
      ...FullType
    }
    directives {
      name
      description
      locations
      args {
        ...InputValue
      }
    }
  }
}

fragment FullType on __Type {
  kind
  name
  description
  fields(includeDeprecated: true) {
    name
    description
    args {
      ...InputValue
    }
    type {
      ...TypeRef
    }
    isDeprecated
    deprecationReason
  }
  inputFields {
    ...InputValue
  }
  interfaces {
    ...TypeRef
  }
  enumValues(includeDeprecated: true) {
    name
    description
    isDeprecated
    deprecationReason
  }
  possibleTypes {
    ...TypeRef
  }
}

fragment InputValue on __InputValue {
  name
  description
  type {
    ...TypeRef
  }
  defaultValue
}

fragment TypeRef on __Type {
  kind
  name
  ofType {
    kind
    name
    ofType {
      kind
      name
      ofType {
        kind
        name
        ofType {
          kind
          name
          ofType {
            kind
            name
            ofType {
              kind
              name
              ofType {
                kind
                name
              }
            }
          }
        }
      }
    }
  }
}
```

**Comment by peteror** [comment]
Thank you! The good news is, the cache works. 
I assumed that it would cache more and have a significant effect on performance (take load from the database and back-end CPU).

**Reply by fowczarek** [reply]
@peteror 
Maybe try this: 
https://docs.saleor.io/docs/3.x/developer/running-saleor/read-replicas

**Reply by peteror** [reply]
@fowczarek thank you, that was the next thing I was looking at :)

**Reply by nigelthomas-mozanta** [reply]
Hi @peteror ,

Can you confirm whether Saleor Redis Cache will work ?

Thanks.

**Reply by peteror** [reply]
I’m still on an older version, but as I see new versions use the cache more extensively.  I haven’t tested it though.

**Comment by timuric** [comment]
We have updated Saleor Cloud pricing, which starts from $45 a month, which provides hassle free updates, backups, autoscale, sandboxes, snapshot management and more. If you are not running forked version of Saleor it is worth to check out.
